partner/partner-management-service/src/main/java/io/mosip/pms/partner/manager/service/impl/PartnerManagementServiceImpl.java

```diff
@@ -583,7 +583,18 @@ public APIKeyGenerateResponseDto generateAPIKey(String partnerId, APIKeyGenerate
 			throw new PartnerManagerServiceException(ErrorCode.LOGGEDIN_USER_NOT_AUTHORIZED.getErrorCode(),
 					ErrorCode.LOGGEDIN_USER_NOT_AUTHORIZED.getErrorMessage());
 		}
-		AuthPolicy validPolicy = authPolicyRepository.findByPolicyName(requestDto.getPolicyName());
+		Optional<Partner> partnerFromDb = partnerRepository.findById(partnerId);
+		if (partnerFromDb.isEmpty()) {
+			auditUtil.setAuditRequestDto(PartnerManageEnum.GENERATE_API_KEY_FAILURE, partnerId, "partnerId");
+			throw new PartnerManagerServiceException(ErrorCode.PARTNER_ID_DOES_NOT_EXIST_EXCEPTION.getErrorCode(),
+					ErrorCode.PARTNER_ID_DOES_NOT_EXIST_EXCEPTION.getErrorMessage());
+		}
+		if (!partnerFromDb.get().getIsActive()) {
+			auditUtil.setAuditRequestDto(PartnerManageEnum.GENERATE_API_KEY_FAILURE, partnerId, "partnerId");
+			throw new PartnerManagerServiceException(ErrorCode.PARTNER_NOT_ACTIVE_EXCEPTION.getErrorCode(),
+					ErrorCode.PARTNER_NOT_ACTIVE_EXCEPTION.getErrorMessage());
+		}
+		AuthPolicy validPolicy = authPolicyRepository.findByPolicyGroupAndName(partnerFromDb.get().getPolicyGroupId(),requestDto.getPolicyName());
```

Comment:
can we name the method more specific to the argument passed - findByPolicyGroupIdAndName

Reply:
Updated the same anusha